### PR TITLE
Feature/add middlewares

### PR DESCRIPTION
**Pull request by kiranbhanushali**
added middleware for check token in  user requests.

### CODE REVIEW

File: middlewares/verify_token.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+const jwt = require("jsonwebtoken");
+const auth_config = require("../config/auth.config");
+require("dotenv").config();
+
+const verify_token = async (req, res, next) => {
+  const req_token = req.get("Authorization");
```
Comment (akshitbansal108) [inline middlewares/verify_token.js:6]: Along with token,  we can also send/receive user id. It is generally done in organisations.

File: controllers/user.controller.js [hunk 2]
```diff
@@ -156,7 +156,7 @@ exports.login = async (req, res) => {
         id: user._id,
```
Comment (kiranbhanushali) [inline controllers/user.controller.js:156]: @akshitbansal108  are you talking about this?.

Comment (akshitbansal108) [reply controllers/user.controller.js:156]: Yes, like when we will send token to frontend on login/register, frontend can store user id along with token and send to the backend in headers whenever Anu api is hit.

**Review by tanu456 — APPROVED** [review]

**Comment by akshitbansal108** [comment]
@kiranbhanushali Please resolve the conflicts and merge it

**Comment by kiranbhanushali** [comment]
done

**Comment by kiranbhanushali** [comment]
done

**Pull request merged**